front/src/services/bedrock.ts

```diff
@@ -0,0 +1,175 @@
+import { EncryptionService } from "@/services/encryption";
+
+export const FileEntrySchema = z.object({
+	path: z.string().regex(/^(\/[A-Za-z0-9-_+.]+)+$/, {
+		message: "Invalid URI format. It should start with '/', contain valid path segments and do not end with a '/'.",
+	}),
+	post_hash: z.string().regex(/^[a-f0-9]{64}$/, {
+		message: "Invalid hash format. It should be a 64-character hexadecimal string.",
+	}),
+});
+
+export const EncryptedFileEntrySchema = z.object({
+	path: z.string().regex(/^[a-f0-9]+$/, { message: "Invalid data, it should be encrypted and in hex format." }),
+	post_hash: z.string().regex(/^[a-f0-9]{64}$/, {
+		message: "Invalid hash format. It should be a 64-character hexadecimal string.",
+	}),
+});
+
+export const FileMetaSchema = z.object({
+	key: z.string().regex(/^[a-f0-9]{64}$/, {
+		message: "Invalid key format. It should be a 64-character hexadecimal string.",
+	}),
+	iv: z.string().regex(/^[a-f0-9]{32}$/, {
+		message: "Invalid IV format. It should be a 32-character hexadecimal string.",
+	}),
+	ipfs_hash: z.string().regex(/^[a-f0-9]{64}$/, {
+		message: "Invalid hash format. It should be a 64-character hexadecimal string.",
+	}),
+});
+
+export const EncryptedFileMetaSchema = z.object({
+	key: z.string().regex(/^[a-f0-9]+$/, { message: "Invalid data, it should be encrypted and in hex format." }),
+	iv: z.string().regex(/^[a-f0-9]+$/, { message: "Invalid data, it should be encrypted and in hex format." }),
+	ipfs_hash: z.string().regex(/^[a-f0-9]+$/, { message: "Invalid data, it should be encrypted and in hex format." }),
+});
+
+export const FileFullInfosSchema = FileEntrySchema.merge(FileMetaSchema);
+export const EncryptedFileFullInfosSchema = EncryptedFileEntrySchema.merge(EncryptedFileMetaSchema);
+
+export const FileEntriesSchema = z.object({
+	files: z.array(FileEntrySchema).default([]),
+});
+
+export const EncryptedFileEntriesSchema = z.object({
+	files: z.array(EncryptedFileEntrySchema).default([]),
+});
+
+const FILE_ENTRIES_AGGREGATE_KEY = "bedrock_file_entries";
+const FILE_POST_TYPE = "bedrock_file";
+
+export type FileEntry = z.infer<typeof FileEntrySchema>;
+export type FileMeta = z.infer<typeof FileMetaSchema>;
+export type EncryptedFileEntry = z.infer<typeof EncryptedFileEntrySchema>;
+export type EncryptedFileMeta = z.infer<typeof EncryptedFileMetaSchema>;
+export type FileFullInfos = z.infer<typeof FileFullInfosSchema>;
+export type DirectoryPath = `${string}/`;
+
+export default class BedrockService {
+	constructor(private alephService: AlephService) {}
+
+	async uploadFiles(directory_path: DirectoryPath, ...files: File[]): Promise<Omit<FileFullInfos, "post_hash">[]> {
+		const uploadedFiles = await this.fetchFileEntries();
+		const results = await Promise.allSettled(
+			files
+				.map((file) => ({ file, path: `${directory_path}${file.name}` }))
+				.filter(({ path }) => {
+					if (this.fileExists(uploadedFiles, path)) {
+						console.error(`File already exists at path: ${path}`);
```

Comment:
not sure about the necessity of a console message + toast but ok

Reply:
To have a log that does not disappear, toasts can be missed